dynamo_db_block.py

```diff
@@ -8,131 +8,46 @@
-        not exist, create it and return that reference.
-
-        Note that if this function does not find the table, it will create it
-        and this creation operation can block for some time (typically ~10s).
-        It will only return the table reference once the table is active and
-        ready to be stored to or read from.
-
-        As a result, it probably makes sense to call this method in a lock
-        for the specific table. Otherwise, simultaneous calls to _get_table
-        could result in multiple table creations.
-
-        Returns:
-            table: A boto table reference
-        """
-        if table_name in self._table_cache:
-            return self._table_cache[table_name]
-
-        table = Table(table_name, connection=self._conn)
+    def execute_signals_query(self, table, signals):
+        """ Save a list of signals to a table reference """
+        with table.batch_write() as batch:
+            for sig in signals:
+                self._save_signal(batch, sig)
 
+    def _save_signal(self, table, signal):
+        """ Save a signal to a DynamoDB table """
         try:
-            num_items = table.count()
-            self._logger.debug("Table {} found - contains {} items".format(
-                table_name, num_items))
-        except JSONResponseError as jre:
-            if 'ResourceNotFoundException' in str(jre):
-                # If we get a resource not found exception, the table must not
-                # exist, so let's create it
-                self._logger.info("Table {} not found - creating it".format(
-                    table_name))
-                table = self._create_table(table_name)
-                self._logger.debug("Table created: {}".format(table))
+            if self._is_valid_signal(signal):
+                table.put_item(data=signal.to_dict())
             else:
-                # We got some other type of exception, raise it since that
-                # wasn't expected
-                raise
+                self._logger.warning(
+                    "Not saving an invalid signal - must contain hash and "
+                    "range keys if specified - {}".format(signal))
         except:
-            self._logger.exception("Unable to determine table reference")
-            return
+            self._logger.exception("Unable to save signal")
 
-        # Cache this reference to the table for later use
-        self._table_cache[table_name] = table
-        return table
+    def _is_valid_signal(self, signal):
+        """ Return true if this signal is valid and can be saved """
+        # A signal has a valid hash if it contains the hash field
+        hash_valid = hasattr(signal, self.hash_key)
+        # A signal has a valid range if it contains the range field or no
+        # range field was specified
+        range_valid = not self.range_key or hasattr(signal, self.range_key)
+
+        return hash_valid and range_valid
 
     def _create_table(self, table_name):
```

Comment:
The reason I didn't like create_table in the base block is because it includes the hash_key and range_key props.